graph-service/page_worker.py

```
@@ -1,302 +1,341 @@
+            # This avoids cartesian product by connecting the patterns
+            query = f"""
+            MATCH (a {{id: $origin}})
+            WITH a
+            MATCH (b {{id: $destination}})
+            MERGE (a)-[r:`{safe_rel_type}`]->(b)
+            """
+            sess.run(query, origin=rel.origin, destination=rel.destination)
+    
+    logger.info(f"Ingested {len(kg.entities)} entities and {len(kg.relations)} relations into Neo4j.")
+
+async def process_text(text: str, source_url: str = "") -> Optional[KGExtractionResult]:
```

Ruff (0.14.7)

Unused function argument: `source_url`

(ARG001)